FirebasePerformance/Sources/Instrumentation/Network/FPRNSURLSessionInstrument.m

```diff
@@ -482,8 +482,15 @@ void InstrumentUploadTaskWithStreamedRequest(FPRNSURLSessionInstrument *instrume
     if (!strongInstrument) {
       ThrowExceptionBecauseInstrumentHasBeenDeallocated(selector, instrumentor.instrumentedClass);
     }
-    typedef NSURLSessionUploadTask *(*OriginalImp)(id, SEL, NSURLRequest *);
-    NSURLSessionUploadTask *uploadTask = ((OriginalImp)currentIMP)(session, selector, request);
+    typedef NSURLSessionUploadTask *(*OriginalImp)(id, SEL, NSURLRequest *, NSData *);
+    // To avoid a runtime warning in Xcode 15, the given `URLRequest`
+    // should have a nil `HTTPBody`. To workaround this, the `HTTPBody` data is removed
+    // and requestData is replaced with it, if it bodyData was `nil`.
+    NSMutableURLRequest *requestWithoutHTTPBody = [request mutableCopy];
+    NSData *requestData = requestWithoutHTTPBody.HTTPBody;
```

Comment:
I'm curious of scenrios where there is data already provided by the developer. Should we add an if condition to do this only when there is external request without any data?

Reply:
Unlike the scenario in the function above, I don't see any other developer provided data in `InstrumentUploadTaskWithStreamedRequest` (or maybe I'm not following your question?)

Reply:
In the request here, we are overwriting the parameter requestData (the 4th parameter) with what we get from the HTTPBody. I was worried about the scenario where the developers calls the UploadTask API with the requestData and we overwrite the value of requestData with the value we have in HTTPBody.

Reply:
After multiple investigation, I see that this is the best approach in resolving the issue. Apparently the extra parameter used is just for getting the upload task objet and so the extra parameter is not causing any issue. Marking the comment resolved and approving the PR.